src/PluginForm/Onsite/PaymentMethodAddForm.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Drupal\commerce_heartland\PluginForm\Onsite;
+
+use Drupal\commerce_payment\PluginForm\PaymentMethodAddForm as BasePaymentMethodAddForm;
+use Drupal\Core\Form\FormStateInterface;
+
+class PaymentMethodAddForm extends BasePaymentMethodAddForm
+{
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function buildCreditCardForm(array $element, FormStateInterface $form_state)
+    {
+        $plugin = $this->plugin;
+
+        $element['#attributes']['class'][] = 'heartland-form';
+
+        $element['#attached']['drupalSettings']['heartland'] = [
+            'publicKey' => $plugin->getPublicKey(),
+        ];
+        $element['#attached']['library'][] = 'commerce_heartland/secure-submit';
+        $element['#attached']['library'][] = 'commerce_heartland/form';
+
+        $element['number'] = [
+            '#type' => 'text',
+            '#required' => true,
+            '#validated' => true,
+            '#markup' => '<div class="form-group">
+                <label for="iframesCardNumber">Card Number:</label>
+                <div class="iframeholder" id="iframesCardNumber"></div>
+                </div>',
+        ];
+
+        $element['expiration'] = [
+            '#type' => 'text',
+            '#markup' => '<div class="form-group">
+                <label for="iframesCardExpiration">Card Expiration</label>
+                <div class="iframeholder" id="iframesCardExpiration"></div>
+                </div>',
+        ];
+
+        $element['security_code'] = [
+            '#type' => 'text',
+            '#markup' => '<div class="form-group">
+                <label for="iframesCardCvv">Card CVV</label>
+                <div class="iframeholder" id="iframesCardCvv"></div>
+                </div>',
+        ];
+
+        $element['token_value'] = [
+            '#type' => 'hidden',
+            '#attributes' => ['id' => 'token_value'],
+        ];
+
+        $element['card_type'] = [
+            '#type' => 'hidden',
+            '#attributes' => ['id' => 'card_type'],
+        ];
+
+        $element['last_four'] = [
+            '#type' => 'hidden',
+            '#attributes' => ['id' => 'last_four'],
+        ];
+
+        $element['exp_month'] = [
+            '#type' => 'hidden',
+            '#attributes' => ['id' => 'exp_month'],
+        ];
+
+        $element['exp_year'] = [
+            '#type' => 'hidden',
+            '#attributes' => ['id' => 'exp_year'],
+        ];
+
+        $element['token_expire'] = [
+            '#type' => 'hidden',
+            '#attributes' => ['id' => 'token_expire'],
+        ];
+
+        return $element;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function validateCreditCardForm(array &$element, FormStateInterface $form_state)
+    {
+        // The JS library performs its own validation.
```

Comment:
Is this empty method intended? Seems like we're preventing the default validations from taking place, but I wanted to verify.

Reply:
Yes, I want to prevent the default validations.